src/Microsoft.AspNet.Mvc.ModelBinding/ModelMetadataTypeAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Microsoft.AspNet.Mvc
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+    public class ModelMetadataTypeAttribute : Attribute
+    {
+        private readonly Type _metadataBuddyType;
```

Comment:
nit: not a big fan of "buddy" or of mixing `buddyType` and `_metadataBuddyType`.  suggest picking one name (e.g. `metadataType` / `_metadataType`) and sticking with it.